### DISCUSSION TITLE
Masking and Animations

### DISCUSSION THREAD

**Discussion by h0bb3** [Comfy-Org/ComfyUI#1605] [Ideas]
Hi,

first of all: great tool and kudos for sharing it as oss!

In my beginner attempts at doing animations with different control nets I found myself wanting a mask. However, I found out that the Convert Image to Mask Node only created the first image as the mask and not the nice batch of images that where actually loaded and needed for my idea.

Digging around in the source I found two minor changes needed in `nodes_masks.py` to fix this:

```python
class ImageToMask:
    @classmethod
    def INPUT_TYPES(s):
        return {
                "required": {
                    "image": ("IMAGE",),
                    "channel": (["red", "green", "blue"],),
                }
        }

    CATEGORY = "mask"

    RETURN_TYPES = ("MASK",)
    FUNCTION = "image_to_mask"

    def image_to_mask(self, image, channel):
        channels = ["red", "green", "blue"]
        mask = image[:, :, :, channels.index(channel)] # this line changed from just using image 0
        return (mask,)
```

The Solid Mask node needs a batch_size attribute (there could be a better way to do this as creating a number of masks that are exactly the same seems a bit bad)

```python
class SolidMask:
    @classmethod
    def INPUT_TYPES(cls):
        return {
            "required": {
                "value": ("FLOAT", {"default": 1.0, "min": 0.0, "max": 1.0, "step": 0.01}),
                "width": ("INT", {"default": 512, "min": 1, "max": MAX_RESOLUTION, "step": 1}),
                "height": ("INT", {"default": 512, "min": 1, "max": MAX_RESOLUTION, "step": 1}),
                "batch_size": ("INT", {"default":1, "min": 1, "step": 1})
            }
        }

    CATEGORY = "mask"

    RETURN_TYPES = ("MASK",)

    FUNCTION = "solid"

    def solid(self, value, width, height, batch_size):
        out = torch.full([batch_size, height, width], value, dtype=torch.float32, device="cpu")
        return (out,)
```

I have tested this using a workflow involving MaskComposite Nodes (subtract, and, & multiply operations) and ConvertMasktoImage Nodes. It seems to work well enough for my needs.

However, I do not have extensive experience of using ComfyUI in general and masks in particular. There is a high probability that more chances are needed and/or that I have broken something. There is also a possibility that I'm doing things in a wrong or awkward way.

Should I rather build this as a custom nodes repo, or is it something that sounds ok to have in the basic mask nodes as per above?

**Comment by comfyanonymous** [comment]
ImageToMask can work with batches of images since yesterday.

**Reply by h0bb3** [reply]
Cool - cant get any faster than a day before I guess :)👍 

I see the implementation was as simple as I thought for that part at least.

I guess the SolidMask will be updated to support batches also?
